fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLAddNameToArgumentIntention.kt

```diff
@@ -0,0 +1,102 @@
+            listOfNotNull(TextRange(0, 0), TextRange(element.textLength, element.textLength))
+        } else {
+            listOf(TextRange(0, element.textLength))
+        }
+    }
+
+    override val inputProvider: HLApplicatorInputProvider<KtValueArgument, Input> = inputProvider { element ->
+        val argumentList = element.parent as? KtValueArgumentList ?: return@inputProvider null
+        val shouldBeLastUnnamed = !element.languageVersionSettings.supportsFeature(LanguageFeature.MixedNamedArgumentsInTheirOwnPosition)
+        if (shouldBeLastUnnamed && element != argumentList.arguments.last { !it.isNamed() }) return@inputProvider null
+
+        // TODO: Cast to KtCallElement (need to handle other subtypes in KtCallResolver mixin)
+        val callElement = argumentList.parent as? KtCallExpression ?: return@inputProvider null
+        val resolvedCall = callElement.resolveCall() as? KtFunctionCall ?: return@inputProvider null
+
+        if (resolvedCall.targetFunction.getSingleCandidateSymbolOrNull()?.hasStableParameterNames != true) {
+            return@inputProvider null
+        }
+
+        val valueParameterSymbol = resolvedCall.argumentMapping[element] ?: return@inputProvider null
+        if (valueParameterSymbol.isVararg) {
+            if (element.languageVersionSettings.supportsFeature(LanguageFeature.ProhibitAssigningSingleElementsToVarargsInNamedForm) &&
+                !element.isSpread) {
+                return@inputProvider null
+            }
+
+            val varargArgumentCount = resolvedCall.argumentMapping.values.count { it == valueParameterSymbol }
+            if (varargArgumentCount != 1) {
+                return@inputProvider null
+            }
+        }
+
+        Input(valueParameterSymbol.name)
+    }
+
+    override fun skipProcessingFurtherElementsAfter(element: PsiElement) = element is KtValueArgumentList ||
+            element is KtContainerNode || super.skipProcessingFurtherElementsAfter(element)
+
+    companion object {
+        val applicator = applicator<KtValueArgument, Input> {
+            familyName(KotlinBundle.lazyMessage("add.name.to.argument"))
+
+            actionName { _, input -> KotlinBundle.message("add.0.to.argument", input.name)  }
+
+            isApplicableByPsi { element ->
+                // Not applicable when lambda is trailing lambda after argument list (e.g., `run {  }`).
+                // Note: Intention IS applicable when lambda is inside an argument list (e.g., `run({  })`).
+                !element.isNamed() && element !is KtLambdaArgument
+            }
+
+            applyTo { element, input ->
+                val expression = element.getArgumentExpression() ?: return@applyTo
+                val name = input.name
+
+                val prevSibling = element.getPrevSiblingIgnoringWhitespace()
+                if (prevSibling is PsiComment && """/\*\s*$name\s*=\s*\*/""".toRegex().matches(prevSibling.text)) {
```

Comment:
Just to double-check. Don't we have any convention regarding where/how to put regex?

Reply:
It's not a constant regex. It includes `$name`. Otherwise I'd put it in a companion object.